test/event-component-factory-mocha.js

```diff
@@ -0,0 +1,84 @@
+// event-component-factory-mocha.js
+
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.should();
+chai.use(chaiAsPromised);
+
+var _ = require('underscore');
+var noflo = require('noflo');
+var componentFactory = require('../components/event-component-factory.js');
+
+describe('event-component-factory', function() {
+    it("should reject undefined definition", function() {
+        return Promise.resolve().then(componentFactory).should.be.rejected;
+    });
+    it("should trigger in port ondata function", function() {
+        var handler;
+        return Promise.resolve({
+            inPorts:{
+                'input':{
+                    ondata: function(payload) {
+                        handler(payload);
```

Comment:
Hard to figure out what 'handler' will eventually be.  Why wasn't it set directly on line 17?